libnmstate/nm/team.py

```diff
@@ -17,14 +17,94 @@
 # along with this program. If not, see <https://www.gnu.org/licenses/>.
 #
 
-from libnmstate.error import NmstateNotImplementedError
+import json
+
+from libnmstate.nm import connection as nm_connection
+from libnmstate.nm import nmclient
+from libnmstate.schema import Interface
 from libnmstate.schema import Team
 
 
-def create_setting(iface_state):
-    setting = None
-    team_config = iface_state.get(Team.CONFIG_SUBTREE)
-    if team_config:
-        raise NmstateNotImplementedError
+TEAMD_JSON_DEVICE = "device"
+
+
+def create_setting(iface_state, base_con_profile):
+    team_setting = None
+    team_config = iface_state.get(Team.CONFIG_SUBTREE, None)
+
+    if not team_config:
+        return None
+
+    if base_con_profile:
+        team_setting = base_con_profile.get_setting_duplicate(
+            nmclient.NM.SETTING_TEAM_SETTING_NAME
+        )
+
+    if not team_setting:
+        team_setting = nmclient.NM.SettingTeam.new()
+
+    if team_config is not None:
+        teamd_config = _convert_team_config_to_teamd_format(
+            team_config, iface_state[Interface.NAME]
+        )
+
+        team_setting.props.config = json.dumps(teamd_config)
+
+    return team_setting
+
+
+def _convert_team_config_to_teamd_format(team_config, ifname):
+    team_config[TEAMD_JSON_DEVICE] = ifname
+
+    team_ports = team_config.get(Team.PORT_SUBTREE, None)
+    team_ports_formatted = {}
+    if team_ports is not None:
+        team_ports_formatted = {
+            port[Team.Port.NAME]: _dict_key_filter(port, Team.Port.NAME)
+            for port in team_ports
+        }
+        team_config[Team.PORT_SUBTREE] = team_ports_formatted
+
+    return team_config
+
+
+def _dict_key_filter(dict_to_filter, key):
+    return dict(filter(lambda elem: elem[0] == key, dict_to_filter.items()))
+
+
+def get_info(device):
+    """
+    Provide the current active teamd values for an interface.
```

Comment:
Please ref a ticket here for the missing kernel state.

Reply:
Sure, done!